File: oxidation/bindings/generator/api/events.py
# Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS

from .common import Variant, ClientHandle


class ClientEvent(Variant):
    class ClientConnectionChanged:
        client: ClientHandle

    class WorkspaceReencryptionNeeded:
        pass

    class WorkspaceReencryptionStarted:
        pass

    class WorkspaceReencryptionEnded:
        pass

File: oxidation/libparsec/crates/client_high_level_api/src/events.rs
// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS

use crate::ClientHandle;

#[derive(Debug)]
pub enum ClientEvent {
    ClientConnectionChanged { client: ClientHandle },
    WorkspaceReencryptionNeeded,
    WorkspaceReencryptionStarted,
    WorkspaceReencryptionEnded,
    // TODO finish me
}
